ZeldaFullEditor/Data/Entrance.cs

```diff
@@ -196,40 +196,127 @@ public class Entrance
-            this.YPosition = (ushort)((ROM.DATA[Constants.entrance_yposition + (entranceID * 2) + 1] << 8) + ROM.DATA[Constants.entrance_yposition + (entranceID * 2)]);
-            this.XPosition = (ushort)((ROM.DATA[Constants.entrance_xposition + (entranceID * 2) + 1] << 8) + ROM.DATA[Constants.entrance_xposition + (entranceID * 2)]);
-            this.CameraX = (ushort)((ROM.DATA[Constants.entrance_camerax + (entranceID * 2) + 1] << 8) + ROM.DATA[Constants.entrance_camerax + (entranceID * 2)]);
-            this.CameraY = (ushort)((ROM.DATA[Constants.entrance_cameray + (entranceID * 2) + 1] << 8) + ROM.DATA[Constants.entrance_cameray + (entranceID * 2)]);
-            this.CameraTriggerY = (ushort)((ROM.DATA[(Constants.entrance_cameraytrigger + (entranceID * 2)) + 1] << 8) + ROM.DATA[Constants.entrance_cameraytrigger + (entranceID * 2)]);
-            this.CameraTriggerX = (ushort)((ROM.DATA[(Constants.entrance_cameraxtrigger + (entranceID * 2)) + 1] << 8) + ROM.DATA[Constants.entrance_cameraxtrigger + (entranceID * 2)]);
-            this.Blockset = ROM.DATA[Constants.entrance_blockset + entranceID];
-            this.Music = ROM.DATA[Constants.entrance_music + entranceID];
-            this.DungeonID = ROM.DATA[Constants.entrance_dungeon + entranceID];
-            this.Floor = ROM.DATA[Constants.entrance_floor + entranceID];
-            this.Door = ROM.DATA[Constants.entrance_door + entranceID];
-            this.LadderBG = ROM.DATA[Constants.entrance_ladderbg + entranceID];
-            this.Scrolling = ROM.DATA[Constants.entrance_scrolling + entranceID];
-            this.ScrollQuadrant = ROM.DATA[Constants.entrance_scrollquadrant + entranceID];
-            this.Exit = (short)((ROM.DATA[Constants.entrance_exit + (entranceID * 2) + 1] << 8) + ROM.DATA[Constants.entrance_exit + (entranceID * 2)]);
-
-            this.CameraBoundaryQN = ROM.DATA[Constants.entrance_scrolledge + 0 + (entranceID * 8)];
-            this.CameraBoundaryFN = ROM.DATA[Constants.entrance_scrolledge + 1 + (entranceID * 8)];
-            this.CameraBoundaryQS = ROM.DATA[Constants.entrance_scrolledge + 2 + (entranceID * 8)];
-            this.CameraBoundaryFS = ROM.DATA[Constants.entrance_scrolledge + 3 + (entranceID * 8)];
-            this.CameraBoundaryQW = ROM.DATA[Constants.entrance_scrolledge + 4 + (entranceID * 8)];
-            this.CameraBoundaryFW = ROM.DATA[Constants.entrance_scrolledge + 5 + (entranceID * 8)];
-            this.CameraBoundaryQE = ROM.DATA[Constants.entrance_scrolledge + 6 + (entranceID * 8)];
-            this.CameraBoundaryFE = ROM.DATA[Constants.entrance_scrolledge + 7 + (entranceID * 8)];
-
-            if (isSpawnPoint)
+            if (!isSpawnPoint)
+            {
+                // if expanded == false
+                if (ROM.DATA[0x07F000] != 00 && ROM.DATA[0x07F001] != 01)
+                {
+                    if (entranceID > 0x84)
```

Comment:
constants for expanded entrance ID upper bound and addresses for flag check

Reply:
Done